File: flutter/user/dotmall/lib/src/features/core/domain/entities/address.dart
class Address {
  final String id;
  final String? primary;
  final String? secondary;
  final String? city;
  final String? state;
  final String? zip;
  final String? country;
  final String? latitude;
  final String? longitude;
  final String? relatedId;
  final String? relatedType;

  const Address({
    required this.id,
    this.primary,
    this.secondary,
    this.city,
    this.state,
    this.zip,
    this.country,
    this.latitude,
    this.longitude,
    this.relatedId,
    this.relatedType,
  });
}

File: adonis/backend/app/Models/Address.ts
import { DateTime } from 'luxon'
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'
import DotBaseModel from '../../dot/models/DotBaseModel'

export default class Address extends DotBaseModel {

  @column()
  public primary: string

  @column()
  public secondary: string

  @column()
  public city: string

  @column()
  public state: string

  @column()
  public zip: string

  @column()
  public country: string

  @column()
  public latitude: string

  @column()
  public longitude: string

  // @column.dateTime({ autoCreate: true })
  // public createdAt: DateTime

  // @column.dateTime({ autoCreate: true, autoUpdate: true })
  // public updatedAt: DateTime
  //
  @column({ serializeAs: null })
  public relatedId: string

  @column({ serializeAs: null })
  public relatedType: string
}
